sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/util/TimeSource.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.sgv2.docsapi.service.util;
+
+/** A simple interface capable of providing current time timestamps in millis and micros. */
```

Comment:
Many frameworks prefer use of nanos as the basic building block: is there a reason for preferring microseconds?
Nanosecond timers are usually not based off the epoch, unlike milliseconds, so conversion between nanos and milliseconds is not necessarily possible.